BootloaderCorePkg/Tools/BuildUtility.py

```diff
@@ -548,24 +538,31 @@ def gen_payload_bin (fv_dir, pld_list, pld_bin, priv_key, brd_name = None):
     # E-payloads container format
     alignment = 0x10
     key_dir  = os.path.dirname (priv_key)
-    pld_list = [('EPLD', '%s' % epld_bin, '', 'RSA2048', '%s' % os.path.basename(priv_key), alignment, 0)]
+    key_type = get_key_type(priv_key)
+    pld_list = [('EPLD', '%s' % epld_bin, '', key_type, '%s' % os.path.basename(priv_key), alignment, 0)]

```

Comment:
we replaced hard-coded RSA2048 to use auth type from parameter, the default parameter SHA2_256 need be updated to RSA2048 to align original behavior.

Reply:
Here we replace RSA2048 with an keytype extracted from PrivateKey file.
For hash alg, it would passed from calling procedure which used self._board.SIGN_HASH_TYPE.

This function wouldn't be used by GenContainer script,.

Reply:
ok. should be updated in GenContainer.py, just added comments there.